testing/scenario_app/bin/android_integration_tests.dart

```diff
@@ -353,6 +371,19 @@ Future<void> _run({
       await Future.wait(pendingComparisons);
     });
 
+    if (contentsGolden != null) {
+      // Check the output here.
+      await step('Check output files...', () async {
+        // TODO(gaaclarke): We should move this into dir_contents_diff.
+        _withTemporaryCwd(contentsGolden, () {
+          final int exitCode = dirContentsDiff(basename(contentsGolden), screenshotPath);
```

Comment:
This prints out the patch relative to //testing/ which is annoying but not the end of the world.  We should clean it up at some point.

Reply:
Oh, in that it uses `print` (or similar) on a failure?

Reply:
The problem is that it is executing `git diff -p` as a subprocess and printing out the output from that.  When you use `git diff -p` the patch that is generated is relative to the CWD.  We don't have the path to the repository at this point.  We just have to path to the golden file.  Another gotcha is that `git diff` will generate absolute paths if you run it from a CWD outside of a repo, which is no good.  I looked for a while to find a flag to `git diff` that would fix it but gave up.  If you are looking for some light reading, don't check out `git help diff` haha